src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/VerifyIotCertificate.java

```diff
@@ -7,67 +7,67 @@
     @Inject
-    public VerifyIotCertificate(IotAuthClient iotAuthClient, CertificateRegistry certificateRegistry) {
+    public VerifyIotCertificate(IotAuthClient iotAuthClient, CertificateRegistry certificateRegistry,
+                                NetworkState networkState) {
         this.iotAuthClient = iotAuthClient;
         this.certificateRegistry = certificateRegistry;
+        this.networkState = networkState;
     }
 
     @Override
     public Result<Boolean> apply(String certificatePem) {
-        Optional<String> certId = iotAuthClient.getActiveCertificateId(certificatePem);
+        // If we think we have network connectivity, then opportunistically go to the
+        // cloud for verification.
+        // If the local registry doesn't have information about the certificate, or if
+        // certificate information is outdated, then also go to the cloud, regardless
+        // of whether we think we're connected.
+        // Else, rely on whatever is in the local registry.
+        boolean verified = false;
 
-        // NOTE: This code will not remove certificates from the registry if they are revoked
-        //  in IoT Core. This is currently okay, as we will fail those connections during the
-        //  TLS handshake.
-        // Eventually, we need to handle offline and cert revocation scenarios. However, that
-        //  will be handled as part of a subsequent PR, as making that change now breaks a lot
-        //  of fragile tests
-        if (certId.isPresent()) {
-            try {
-                Certificate clientCertificate;
-                Optional<Certificate> cert = certificateRegistry.getCertificateFromPem(certificatePem);
+        try {
+            Optional<Certificate> cert = certificateRegistry.getCertificateFromPem(certificatePem);
+            if (cert.isPresent() && cert.get().isActive()) {
+                verified = true;
+            }
 
+            if (networkState.getConnectionStateFromMqtt() == NetworkState.ConnectionState.NETWORK_UP || !verified) {
+                cert = iotAuthClient.getIotCertificate(certificatePem);
                 if (cert.isPresent()) {
-                    clientCertificate = cert.get();
-                } else {
-                    clientCertificate = certificateRegistry.createCertificate(certificatePem);
+                    verified = cert.get().isActive();
                 }
+            }
 
-                clientCertificate.setStatus(Certificate.Status.ACTIVE);
-                certificateRegistry.updateCertificate(clientCertificate);
-            } catch (InvalidCertificateException e) {
-                // This should never happen
-                logger.atError()
-                        .kv("CertificateID", certId.get())
-                        .log("Unable to create Certificate object from client certificate, despite having received a "
-                                + "valid certificate ID from IoT Core");
+            if (!cert.get().isActive()) {
```

Comment:
this is broken. we're never storing the cert so session creation fails later. Need to fix